ssl/ssl_ciph.c

```diff
@@ -1971,3 +1971,61 @@ int SSL_CIPHER_get_digest_nid(const SSL_CIPHER *c)
         return -1;
     return ssl_cipher_table_mac[i].nid;
 }
+
+int SSL_CIPHER_get_key_exchange(const SSL_CIPHER *c)
```

Comment:
I left as int, since compilers will change the enumerated type based on the values of the enumerations. Given the range of values, the enumerations can be represented by a char. I suppose I could add a max value, and then the type can be forced to be a 'short' or an 'int'.